bot/cogs/moderation/infractions.py

```diff
@@ -244,7 +248,14 @@ def __init__(self, bot: Bot):
 
         self.mod_log.ignore(Event.member_remove, user.id)
 
-        action = ctx.guild.ban(user, reason=reason, delete_message_days=0)
+        if len(reason) > 512:
+            log.info("Ban reason is longer than 512 characters. Reason will be truncated for Audit Log.")
+
+        action = ctx.guild.ban(
+            user,
+            reason=textwrap.shorten(reason, width=512, placeholder=" ..."),
+            delete_message_days=0
+        )
```

Comment:
If you put the reason in a separate variable above you could avoid having to break up the function arguments for `ban()`.